sktime/forecasting/tests/test_naive.py

```diff
@@ -401,3 +401,18 @@ def test_naive_predict_interval_against_R_naive(strategy, sp, lower, upper):
     expected[(0, coverage, "upper")] = upper
 
     pd.testing.assert_frame_equal(y_pred_ints, expected)
+
+
+def test_naive_sp_greater_1_not_nan():
+    sample_dates = pd.date_range(start="2001-01-01", periods=30, freq="2D")
```

Comment:
Thank you. I tested it and it didn't worked with your specified indices.

Reply:
is this the failing case?

Reply:
Thanks for checking the bad cases, I continuously face issues with these frequencies at work, so guessed as such.

If it's okay with you, will you please parametrise the test for these frequencies as well? I think adding something like this and taking the `frequency` as argument to the test function will be enough.

```python
@pytest.mark.parametrize("frequency", ["D", "W", "W-TUE", "MS", "ME"])
```

Reply:
I added the test with parameters. But I think "ME" is no valid frequency. And "MS" leads to an exception that is not related to this issue.

Reply:
> I think "ME" is no valid frequency

I can see `ME` here though. May be it's because of pandas version.

https://pandas.pydata.org/docs/user_guide/timeseries.html#offset-aliases

---

Another small request, and I don't know if it matters or not, can you please add `2W` along with `W` and etc. in the tests? This is irrespective of the final fix you and @fkiraly decides.

(I am not able to follow this discussion becuase of so many problems and solutions you are trying.)

---

Is something like this useful in any way?

```python
>>>
>>> a = pd.Timestamp(year=2024, month=1, day=31)
>>>
>>> a + pd.offsets.MonthEnd(n=1)
Timestamp('2024-02-29 00:00:00')
>>>
>>> a + pd.offsets.MonthEnd(n=-2)
Timestamp('2023-11-30 00:00:00')
>>>
```

(I am sorry if it's completely unrelated, I am not familiar with how exactly forecasting horizon works and it seems too complicated to learn quickly.)